### ISSUE TITLE
Refactor Log class

### ISSUE THREAD

**Issue by kenorb** [EA31337/EA31337-classes#264]
We need to refactor Log class, so inherited logs can be managed in the easier way.

E.g.
```

Ref<Log> = new Log();

log.add(new Log);

class Log : public Object
{
  WeakRef<Log> parent_log;
  List<Ref<Log>> child_logs;
}
```

> The question is whether we want to make life easier for ourselves. as far as performance is concerned, the only bootleneck would be to create an object that holds the reference counters, but using a linked list you can make it ultra fast. in C++ you can override the `new operator`, then you allocate such a structure of counters right at the beginning of the object, so there is no allocation overhead, except that the object is 14 bytes bigger.

> The advantage is that if you have such a collection of objects, you can decide for yourself whether the objects you are holding are to exist (Ref) or can but don't have to (WeakRef).

> Then you iterate over the logs and check:

```
if (log.ObjectExists()) {
  ...
}
```

**Comment by kenorb** [comment]
Known bugs with old implementation:

- #276

**Comment by kenorb** [comment]
Fixed in #269/#279.
